src/mixins/java/org/spongepowered/common/mixin/core/server/level/ServerPlayerMixin.java

```diff
@@ -830,6 +831,30 @@ public void sendMessage(final net.minecraft.network.chat.Component p_241151_1_,
             }
         }
     }
+
+    @Inject(method = "setPlayerInput", at = @At("HEAD"), cancellable = true)
+    public void onSetPlayerInput(final float sway, final float surge, final boolean jumping, final boolean dismount, final CallbackInfo ci) {
+        if(isPassenger()) {
+            if(sway == this.xxa && surge == this.zza && jumping == this.jumping) {
+                return;
+            }
+
+            final SteerVehicleEvent event = SpongeEventFactory.createSteerVehicleEvent(
+                    PhaseTracker.getCauseStackManager().currentCause(),
+                    (org.spongepowered.api.entity.living.player.server.ServerPlayer) this,
+                    jumping,
+                    surge,
+                    sway
+            );
+
+            Sponge.eventManager().post(event);
+
+            if(event.isCancelled()) {
+                ci.cancel();
+            }
```

Comment:
You could replace this with 

```suggestion
            return !Sponge.eventManager().post(event);
```